app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaScreen.kt

```diff
@@ -334,6 +339,23 @@ fun AgendaScreenContent(
         }
     }
 
+    // Timer to update the time needle every second
+    val zonedDateTimeNow = remember { mutableStateOf(ZonedDateTime.now()) }
+    val timer = remember { Timer() }
+    val timerTask = remember { object : TimerTask() {
+        override fun run() {
+            zonedDateTimeNow.value = ZonedDateTime.now()
+        }
+    } }
+    DisposableEffect(true) {
+        timer.scheduleAtFixedRate(timerTask, 0, 1000)
+
+        onDispose {
+            timerTask.cancel()
+            timer.cancel()
+        }
+    }
```

Comment:
Any reason you didn't implement this UI state logic in the VM?

Reply:
Im not sure what you mean here... This is a ticker to refresh the list and time needle. 

How should the VM do this? Isnt this the responsibility of the UI layer?

Reply:
In the end you're updating UI state here what is usually the responsibility of the ViewModel unless there's a good reason to do it in the UI. The same could be done in the ViewModel with a timer which you cancel in `onCleared()`

Reply:
OK I have refactored for this.